core/state.go

```diff
@@ -118,7 +119,7 @@ func (s *State) putStateStorage(state *trie.Trie) error {
 // updated if an error is encountered during the operation. If update's
 // old or new root does not match the state's old or new roots,
 // [ErrMismatchedRoot] is returned.
-func (s *State) Update(update *StateUpdate) error {
+func (s *State) Update(update *StateUpdate, declaredClasses map[felt.Felt]*Class) error {
```

Comment:
We actually need to create a new `DeclaredContract` (we can change the name) type because `Class` won't help us fulfill any RPC endpoint. See [here](https://playground.open-rpc.org/?uiSchema%5BappBar%5D%5Bui:splitView%5D=false&schemaUrl=https://raw.githubusercontent.com/starkware-libs/starknet-specs/master/api/starknet_api_openrpc.json&uiSchema%5BappBar%5D%5Bui:input%5D=false&uiSchema%5BappBar%5D%5Bui:darkMode%5D=true&uiSchema%5BappBar%5D%5Bui:examplesDropdown%5D=false).

We need the Abi (not present here), Entrypoints, and program(not here). See the [previous implementation](https://github.com/NethermindEth/juno/blob/jelilat/implement-class-storage/core/state_update.go#L37-L41). It also contains program compression and so on.

Reply:
The major purpose of `Class` type is to help us calculate the class hash.

Reply:
I agree that we need to modify the class struct to contain other information that users can optionally include in their database (such as ABIs and programs). However, I think that would be outside the scope of this PR, where we're just trying to store classes in the database.

Reply:
I agree with @joshklop. We can worry about how best to serve Classes through the rpc in the next phase.